ui/packages/atlasmap-core/src/contracts/documents/csv.ts

```diff
@@ -66,3 +66,115 @@ export interface ICsvField extends IField {
 export interface ICsvFields {
   csvField: ICsvField[];
 }
+
+export interface IParameterOption {
+  label: string;
+  value: string;
+}
+
+export interface IParameter {
+  name: string;
+  label: string;
+  value: string;
+  boolean?: boolean;
+  options?: IParameterOption[];
+  enabled?: boolean;
+  required?: boolean;
+}
+
+export class ICsvParameterOptions {
```

Comment:
ah maybe even
```typescript
const CSV_PARAMETER_OPTIONS: IParameter[] = [
  {
    name: 'format',
...
```

Reply:
I guess if we're going to eventually get this from the BE then having a class there will mean less changes going forward.  But we can certainly go with a const rather than static class.  wdyt?  I'll squash once all is decided.

Reply:
I'd vote on const - I think retrieving from backend won't be implemented soon.

Reply:
ok - using const

Reply:
const arrays are still mutable in typescript interestingly enough.  Consequently with
 `const CSV_PARAMETER_OPTIONS: IParameter[] = [...`
you can legally set `CSV_PARAMETER_OPTIONS[3]`.  This resulted in cross-dialog element corruption where the constant array from csv,ts gets changed.  I even tried
`const CSV_PARAMETER_OPTIONS: ReadonlyArray<IParameter> = [` and it still got munged.
I think the best way is to return a fresh static class array.  That works fine in all cases.

Reply:
`as const` might help? - https://stackoverflow.com/questions/43120022/i-want-totally-immutable-object-in-ts

Reply:
We can't use `as const` with `'IParameter[]` since it's a mutable type.  We'd have to create another local immutable type reference or cast `as IParameter` on all references.  I'd like to keep as is now if possible...

Reply:
I mean isn't it possible to have `const CSV_PARAMETER_OPTIONS: IParameter[] = [..] as const`?

Reply:
No - you get this error:
`const CSV_PARAMETER_OPTIONS: IParameter[]`
`The type 'readonly [{ readonly name: "format"; readonly label: "CSV File Format"; readonly value: "Default"; readonly options: readonly [{ readonly label: "Default"; readonly value: "Default"; }, { readonly label: "Excel"; readonly value: "Excel"; }, ... 9 more ..., { ...; }]; readonly required: true; readonly enabled: true; ...' is 'readonly' and cannot be assigned to the mutable type 'IParameter[]'.ts(4104)`

Reply:
So it doesn't enforce on existing types... pity. LGTM then, can you squash the commits?